language/vm/src/file_format.rs

```diff
@@ -1144,8 +1144,15 @@ impl Bytecode {
 
     /// Return the successor offsets of this bytecode instruction.
     pub fn get_successors(pc: CodeOffset, code: &[Bytecode]) -> Vec<CodeOffset> {
+        checked_precondition!(
+            pc <= u16::max_value() - 2 && (pc as usize) < code.len(),
```

Comment:
I was wondering why pc is allowed to be code.len() - 1, so I looked at line 1162 below and it smells bad to me. A malformed binary could result in a vector of size one being returned to a call site that expects two successors. Presumably that call site will fail, but failing at this point might be better for debugging.

Reply:
Thanks for pointing that out. I'll make a note of it.